glyphs-reader/src/font.rs

```diff
@@ -347,359 +352,275 @@ struct NumberName {
+                let key = key
+                    .parse::<f64>()
+                    .or_else(|_| key.parse::<i64>().map(|i| i as f64))
+                    .ok()?;
+                let val = value.as_f64()?;
+                Some((key.into(), val.into()))
+            })
+            .collect()
     }
 
-    fn glyph_order(&self) -> Option<&Vec<SmolStr>> {
-        let Some(CustomParameterValue::GlyphOrder(names)) = self.get("glyphOrder") else {
-            return None;
-        };
-        Some(names)
+    fn as_axis_mappings(&self) -> Option<Vec<AxisMapping>> {
+        self.as_dict()?
+            .iter()
+            .map(|(tag, mappings)| {
+                let user_to_design = mappings.as_mapping_values()?;
+                Some(AxisMapping {
+                    tag: tag.to_string(),
+                    user_to_design,
+                })
+            })
+            .collect()
     }
 
-    fn virtual_masters(&self) -> impl Iterator<Item = &Vec<AxisLocation>> {
-        self.0.iter().filter_map(|(name, value)| {
-            if name == "Virtual Master" {
-                let CustomParameterValue::VirtualMaster(locations) = value else {
-                    panic!("Virtual Master parameter has wrong type!");
-                };
-                return Some(locations);
-            }
-            None
-        })
+    fn as_virtual_master(&self) -> Option<BTreeMap<String, OrderedFloat<f64>>> {
+        self.as_array()?
+            .iter()
+            .map(Plist::as_axis_location)
+            .map(|loc| (loc.map(|loc| (loc.axis_name, loc.location))))
+            .collect()
     }
 
-    fn fs_type(&self) -> Option<&Vec<i64>> {
-        let Some(CustomParameterValue::FsType(bits)) = self.get("fsType") else {
-            return None;
-        };
-        Some(bits)
+    fn as_fs_type(&self) -> Option<u16> {
+        Some(self.as_vec_of_ints()?.iter().map(|bit| 1 << bit).sum())
     }
 
-    fn unicode_range(&self) -> Option<&Vec<i64>> {
-        let Some(CustomParameterValue::UnicodeRange(bits)) = self.get("unicodeRanges") else {
-            return None;
-        };
-        Some(bits)
+    fn as_unicode_code_ranges(&self) -> Option<BTreeSet<u32>> {
+        let bits = self.as_vec_of_ints()?;
+        Some(bits.iter().map(|bit| *bit as u32).collect())
     }
 
-    fn codepage_range(&self) -> Option<&Vec<i64>> {
-        let Some(CustomParameterValue::CodepageRange(bits)) = self.get("codePageRanges") else {
-            return None;
-        };
-        Some(bits)
+    fn as_codepage_bits(&self) -> Option<BTreeSet<u32>> {
+        let bits = self.as_vec_of_ints()?;
+        bits.iter()
+            .map(|b| codepage_range_bit(*b as _))
+            .collect::<Result<_, _>>()
+            .ok()
     }
+}
 
-    fn panose(&self) -> Option<&Vec<i64>> {
+impl RawCustomParameters {
+    ////convert into the parsed params for a top-level font
+    fn cook(&self) -> Result<CustomParameters, Error> {
```

Comment:
If I see `x.cook()` in code I wouldn't know what it did, can we get a more descriptive name? - for example, if I saw `.create_custom_parameters()?` that would be far clearer to me.